src/Formatter/NMEAFormatter.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2023 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "Units/Units.hpp"
+#include "NMEAFormatter.hpp"
+
+void
+FormatGPRMC(char *buffer, size_t size, const NMEAInfo &info) noexcept
+{
+  const GeoPoint location = info.location_available
+    ? info.location
+    : GeoPoint::Zero();
+
+  const BrokenDateTime now = info.time_available &&
+    info.date_time_utc.IsDatePlausible()
+    ? info.date_time_utc
+    : BrokenDateTime::NowUTC();
```

Comment:
The fallback to the current time stamp seems surprising and wrong to me, it's not documented, and the system clock may be wrong. Think of a Kobo or a Raspi that has no battery-buffered real-time clock.
I say this library should not mandate any such fallback, it should only take values from its parameters and never do any system calls.